calour/dsfdr.py

```diff
@@ -272,15 +272,18 @@ def dsfdr(data, labels, transform_type='rankdata', method='meandiff',
             index = np.nonzero(data[i, :])
             label_nonzero = labels[index]
             sample_nonzero = data[i, :][index]
+            if len(sample_nonzero) == 0:
+                continue
             if method == 'nonzerospearman':
                 sample_nonzero = sp.stats.rankdata(sample_nonzero)
                 label_nonzero = sp.stats.rankdata(label_nonzero)
             sample_nonzero = sample_nonzero - np.mean(sample_nonzero)
             label_nonzero = label_nonzero - np.mean(label_nonzero)
             tstat[i] = np.dot(sample_nonzero, label_nonzero)
             t[i] = np.abs(tstat[i])
+            if np.std(sample_nonzero) == 0:
+                continue
```

Comment:
less convenient since sample_nonzero is modified after the first if and before this if... and i think also will be less clear to understand if joined.